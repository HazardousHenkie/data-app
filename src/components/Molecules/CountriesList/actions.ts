import { action } from 'typesafe-actions'

import ActionTypes from './constants'

export const getCountriesData = () => action(ActionTypes.GET_COUNTRIES_DATA)
export const getCountriesDataSuccess = (countriesData: []) =>
    action(ActionTypes.GET_COUNTRIES_DATA_SUCCESS, countriesData)
export const getCountriesDataError = (error: Error) =>
    action(ActionTypes.GET_COUNTRIES_DATA_ERROR, error)
