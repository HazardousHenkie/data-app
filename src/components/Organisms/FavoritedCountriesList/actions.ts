import { action } from 'typesafe-actions'

import ActionTypes from './constants'

import { CountryState } from './types'

const setSelectedCountry = (country: CountryState['country']) =>
    action(ActionTypes.GET_FAVORITED_COUNTRIES, country)

export default setSelectedCountry
