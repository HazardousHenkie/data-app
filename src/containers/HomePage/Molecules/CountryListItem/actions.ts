import { action } from 'typesafe-actions'

import ActionTypes from './constants'

import CountryState from './types'

const setSelectedCountry = (country: CountryState['country']) =>
    action(ActionTypes.SET_SELECTED_COUNTRY, country)

export default setSelectedCountry
