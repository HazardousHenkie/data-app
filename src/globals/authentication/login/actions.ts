import { action } from 'typesafe-actions'

import AuthenticationState from '../types'
import ActionTypes from './constants'

export const loginRequest = (loginToken: string) =>
    action(ActionTypes.LOGIN_REQUEST, loginToken)
export const loginSuccess = (user: AuthenticationState['user']) =>
    action(ActionTypes.LOGIN_SUCCESS, user)
export const loginError = (error: Error | boolean) =>
    action(ActionTypes.LOGIN_ERROR, error)
