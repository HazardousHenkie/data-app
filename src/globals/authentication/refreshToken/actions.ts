import { action } from 'typesafe-actions'

import AuthenticationState from '../types'
import ActionTypes from './constants'

export const getRefreshTokenRequest = (id: string) =>
    action(ActionTypes.GET_REFRESH_TOKEN_REQUEST, id)
export const getRefreshTokenSuccess = (user: AuthenticationState['user']) =>
    action(ActionTypes.GET_REFRESH_TOKEN_SUCCESS, user)
export const getRefreshTokenError = (error: Error | boolean) =>
    action(ActionTypes.GET_REFRESH_TOKEN_ERROR, error)
