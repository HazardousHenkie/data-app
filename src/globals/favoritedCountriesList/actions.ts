import { action } from 'typesafe-actions'

import { ResponseError } from 'utils/request'
import ActionTypes from './constants'

import FavoritedCountriesState from './types'

export const getFavoritedCountries = () =>
    action(ActionTypes.GET_FAVORITED_COUNTRIES)

export const setFavoritedCountries = (
    favoritedCountries: FavoritedCountriesState['countries']
) => action(ActionTypes.SET_FAVORITED_COUNTRIES, favoritedCountries)

export const getFavoritedCountriesError = (error: ResponseError) =>
    action(ActionTypes.GET_FAVORITED_COUNTRIES_ERROR, error)
