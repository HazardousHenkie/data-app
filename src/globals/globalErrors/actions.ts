import { action } from 'typesafe-actions'

import ActionTypes from './constants'

export const getErrors = () => action(ActionTypes.GET_ERRORS)

export const setError = (error: Error) => action(ActionTypes.SET_ERROR, error)

export const removeError = () => action(ActionTypes.REMOVE_ERROR)

export const setErrors = (errors: Error[]) =>
    action(ActionTypes.SET_ERRORS, errors)
