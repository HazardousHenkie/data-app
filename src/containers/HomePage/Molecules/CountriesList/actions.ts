import { action } from 'typesafe-actions'

import { CountryInterface } from 'containers/HomePage/Molecules/CountryListItem/types'
import ActionTypes from './constants'

export const getCountriesData = () => action(ActionTypes.GET_COUNTRIES_DATA)
export const getCountriesDataSuccess = (countriesData: CountryInterface[]) =>
    action(ActionTypes.GET_COUNTRIES_DATA_SUCCESS, countriesData)
export const getCountriesDataError = (error: Error) =>
    action(ActionTypes.GET_COUNTRIES_DATA_ERROR, error)
