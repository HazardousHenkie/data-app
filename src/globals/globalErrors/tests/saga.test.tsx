import { put } from 'redux-saga/effects'

import { ActionType as typeSafeAction } from 'typesafe-actions'
import ActionTypes from '../constants'

import { setError, setErrors } from '../actions'
import globalAddErrorSaga from '../saga'

describe('globalAddErrorSaga Saga', () => {
    let errorsGenerator: Generator
    const params = {
        type: ActionTypes.SET_ERROR,
        payload: Error('test')
    }
    beforeEach(() => {
        errorsGenerator = globalAddErrorSaga(
            params as typeSafeAction<typeof setError>
        )

        const selectDescriptor = errorsGenerator.next().value
        expect(selectDescriptor).toMatchSnapshot()
    })

    it('should dispatch the setErrors action if it selects the errors successfully', () => {
        const errors = [params.payload]
        const putDescriptor = errorsGenerator.next(errors).value

        expect(putDescriptor).toEqual(
            // eslint-disable-next-line redux-saga/no-unhandled-errors
            put(setErrors([...errors, params.payload]))
        )
    })

    it('should call the setError action if the response errors', () => {
        const response = new Error('Some error')
        const putDescriptor = errorsGenerator.throw(response).value
        // eslint-disable-next-line redux-saga/no-unhandled-errors
        expect(putDescriptor).toEqual(put(setError(response)))
    })
})
