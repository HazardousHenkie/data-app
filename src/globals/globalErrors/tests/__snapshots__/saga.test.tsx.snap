// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`globalAddErrorSaga Saga should call the setError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`globalAddErrorSaga Saga should dispatch the setErrors action if it selects the errors successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;
